src/Library/demos/Camera/main.js

```diff
@@ -0,0 +1,76 @@
+import Gtk from "gi://Gtk";
+import Xdp from "gi://Xdp";
+import XdpGtk from "gi://XdpGtk4";
+import Gio from "gi://Gio";
+import GObject from "gi://GObject";
+import GLib from "gi://GLib";
+import Gst from "gi://Gst";
+
+const portal = new Xdp.Portal();
+const parent = XdpGtk.parent_new_gtk(workbench.window);
+const video = workbench.builder.get_object("video");
+
+const button = workbench.builder.get_object("button");
+async function onClicked() {
+  if (portal.is_camera_present) {
+    const pwRemote = await portal.open_pipewire_remote_for_camera();
+    print("Pipewire remote opened for camera");
+    Gst.init(null);
+
+    // Create the pipeline
+    const pipeline = new Gst.Pipeline();
+
+    // Create elements
+    const source = Gst.ElementFactory.make("pipewiresrc", "source");
+    const videoConvert = Gst.ElementFactory.make(
+      "videoconvert",
+      "videoConvert",
+    );
+    const videoSink = Gst.ElementFactory.make("gtksink", "videoSink");
+
+    // Set properties
+    source.set_property("path", pwRemote); // pwRemote is the pipewiresrc obtained from libportal
+
+    // Create the bin
+    const bin = new Gst.Bin();
+    bin.add(source);
+    bin.add(videoConvert);
+    pipeline.add(bin);
```

Comment:
So you create a bin but don't link it to the sink and you're not setting up ghost pads, not gonna work :)
A bin is not needed, you can add source and videoconvert to the pipeline directly, it's a bin already.